tooling-mp-jwt-auth-tool/src/main/java/org/jboss/eap/qe/microprofile/jwt/auth/tool/JweHelper.java

```diff
@@ -0,0 +1,289 @@
+package org.jboss.eap.qe.microprofile.jwt.auth.tool;
+
+import java.net.URI;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.util.List;
+import java.util.Map;
+
+import org.jose4j.base64url.Base64Url;
+import org.jose4j.jwa.AlgorithmConstraints;
+import org.jose4j.jwe.ContentEncryptionAlgorithmIdentifiers;
+import org.jose4j.jwe.JsonWebEncryption;
+import org.jose4j.jwe.KeyManagementAlgorithmIdentifiers;
+import org.jose4j.jws.AlgorithmIdentifiers;
+import org.jose4j.jws.JsonWebSignature;
+import org.jose4j.jwt.consumer.JwtConsumer;
+import org.jose4j.jwt.consumer.JwtConsumerBuilder;
+import org.jose4j.jwt.consumer.JwtContext;
+
+/**
+ * Utility wrapper for the jose4j library for creating and reading JWE containing a nested (signed and encrypted) JWT
+ *
+ * Inspired by
+ * https://bitbucket.org/b_c/jose4j/wiki/JWT%20Examples#markdown-header-producing-and-consuming-a-nested-signed-and-encrypted-jwt
+ *
+ * @author tborgato
+ */
+public class JweHelper {
+
+    private JwsHelper jwsHelper;
+    private RsaKeyHelper rsaHelper;
+    private EcKeyHelper ecHelper;
+
+    private JwsHelper getJwsHelper() {
+        if (jwsHelper == null)
+            jwsHelper = new JwsHelper();
+        return jwsHelper;
+    }
+
+    private RsaKeyHelper getRsaHelper() {
+        if (rsaHelper == null)
+            rsaHelper = new RsaKeyHelper();
+        return rsaHelper;
+    }
+
+    private EcKeyHelper getEcHelper() {
+        if (ecHelper == null)
+            ecHelper = new EcKeyHelper();
+        return ecHelper;
+    }
+
+    public JsonWebEncryption generateJwe(
```

Comment:
Method with more than 4 arguments is seen as bad practice. IMHO a builder pattern would be very useful here.

Reply:
@honza-kasik good point, do you want me to change it in this PR?

Reply:
@honza-kasik added TODO for next PR